Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1977,3 +1985,62 @@
     cairo_path_destroy(caPath);
     return (CGPathRef)copyPath;
 }
+
+/**
+ * Helper method to render text with the given DWRITE_GLYPH_RUN.
+ * Today we create a IWICBitmap with the underlying XAML Writable Bitmap raw bytes and rasterize to it before copying it back to the
+ * original buffer.
+ * TODO::
+ *    1. This method today does not handle DPI scaling. Investigate what needs to be one here to support it
+ *    2. There is double copy happening here - first when the IWICBitmap is created and later the rasterized image is copied back
+ *       Implementing our own IWICBitmap interface should fix this issue.
+ *
+ * @parameter glyphRun DWRITE_GLYPH_RUN object to render
+ */
+void CGContextCairo::CGContextDrawGlyphRun(const DWRITE_GLYPH_RUN* glyphRun) {
```

Comment:
nit: this function body could use a lot more comments - I can grasp the surface of what's happening, but I get the feeling that there's a lot of subtle stuff that someone else wouldn't pick up on first-time read?